benchmark/simulate_hgts.py

```diff
@@ -0,0 +1,627 @@
+                substitute_genes[0], substitute_genes[1]))
+        # rename recipient orthologous gene to donor's
+        hgt_gene = "%s_hgt_o" % gene_donor_label
+        genes_recip[hgt_gene] = genes_recip.pop(gene_recip_label)
+        # replace recipient gene (translated sequence) with donor's
+        genes_recip[hgt_gene][0] = genes_donor[gene_donor_label][0]
+        # update end position of HGT gene (as it can be shorter/longer than
+        # the recipient gene replaced)
+        genes_recip[hgt_gene][2] =\
+            genes_recip[hgt_gene][1] + len(genes_recip[hgt_gene][0])*3
+        # replace recipient gene (nucleotide format) with donor's
+        start_pos_recip, end_pos_recip, strand_recip =\
+            genes_recip[hgt_gene][1:]
+        start_pos_donor, end_pos_donor, strand_donor =\
+            genes_donor[gene_donor_label][1:]
+        seq_recip = Sequence(str(seq_recip[:start_pos_recip]) +
+                             str(seq_donor[start_pos_donor:end_pos_donor]) +
+                             str(seq_recip[end_pos_recip:]))
+        if strand_recip != strand_donor:
+            genes_recip[hgt_gene][3] = genes_donor[gene_donor_label][3]
+        # write HGTs to log file
+        log_f.write("o\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n" % (
+            gene_donor_label,
+            start_pos_donor,
+            end_pos_donor,
+            gene_recip_label,
+            hgt_gene,
+            start_pos_recip,
+            end_pos_recip,
+            strand_donor))
+    return seq_recip
+
+
+def simulate_novel_acq(genes_donor,
+                       seq_donor,
+                       genes_recip,
+                       seq_recip,
+                       orthologous_rep_prob,
+                       percentage_hgts,
+                       log_f):
+    """Simulate novel gene acquisition HGT.
+
+    Parameters
+    ----------
+    genes_donor: dictionary
+    seq_donor: skbio.sequence.Sequence
+    genes_recip: dictionary
+    seq_recip: skbio.sequence.Sequence
+    orthologous_rep_prob: float
+    percentage_hgts: float
+    log_f: file descriptor
+
+    Returns
+    -------
+    seq_recip: skbio.sequence.Sequence
+        recipient genome sequence with HGTs
+
+    Notes
+    -----
+    Algorithm:
+        1. choose random location in recipient genome where to insert a gene
+           (randomly chosen from list of donor genes)
+        2. use gene (recipient) positioning array to locate an open region
+           (that doesn't include an existing gene) near the random location
+           to insert the new gene (we want to avoid gene overlap so that
+           compositional methods can clearly pick out individual coding
+           genes)
+        3. insert new gene, record existance in gene positioning array
+    """
+    num_hgts = int(percentage_hgts*(1-orthologous_rep_prob)*len(genes_recip))
+    if num_hgts < 1:
+        num_hgts = 1
+    # create recipient genome gene positioning array
```

Comment:
```gene_positions = [(0, 0), (len(seq_recip), len(seq_recip))]``` is a good suggestion. However, the values of ```gene_recip``` are actually lists containing 4 variables - the gene sequence, start position, end position and the strand (+ or -).